.github/workflows/pr.yaml

```
@@ -1,18 +1,32 @@
 name: PR Test operator
 
 on:
-  pull_request:
+  pull_request_target:
+    types: [opened, synchronize, reopened, ready_for_review]
     branches: 
     - main
     - rhdh-1.[0-9]+
     - 1.[0-9]+.x
     - release-1.[0-9]+
 
 jobs:
+  authorize:
+    # The 'external' environment is configured with the odo-maintainers team as required reviewers.
+    # All the subsequent jobs in this workflow 'need' this job, which will require manual approval for PRs coming from external forks.
+    # see list of approvers in OWNERS file
+    environment:
+      ${{ (github.event.pull_request.head.repo.full_name == github.repository ||
+        contains(fromJSON('["gazarenkov","nickboldt","rm3l","openshift-cherrypick-robot"]'), github.actor)) && 'internal' || 'external' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: approved
+        run: |
+          echo "✓"
+
   pr-validate:
     name: PR Validate
     runs-on: ubuntu-latest
-    steps:
+    steps:          
```

```suggestion
    needs: authorize
    steps:
```

Otherwise, there is no point in having the `authorize` job above.